packages/typespec-python/test/mock_api_tests/asynctests/test_azure_core_traits_async.py

```diff
@@ -36,3 +39,42 @@ async def test_get(client: TraitsClient):
     assert header["ETag"] == "11bdc430-65e8-45ad-81d9-8ffa60d55b59"
     assert header["bar"] == "456"
     assert header["x-ms-client-request-id"] == "test-id"
+
+
+@pytest.mark.asyncio
+async def test_repeatable_action(client: TraitsClient):
+    result, header = await client.repeatable_action(
+        id=1,
+        body=UserActionParam(user_action_value="test"),
+        cls=lambda x, y, z: (y, z),
```

Comment:
It doesn't seem you're using this lambda?

Please pass the kwarg raw_request_hook, to confirm that the headers have been passed. It's technically tested by the mock api, but I think it helps clarify that the intent of this test is to confirm the SDK is building those values.
https://github.com/Azure/azure-sdk-for-python/blob/ee772c2e222071154bdd212d43b470ecdb97214f/sdk/core/azure-core/azure/core/pipeline/policies/_custom_hook.py#L48-L51

Reply:
The lambda is used for `header` return. I've added `raw_reqest_hook` check.